cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1066,6 +1066,106 @@ VersionedItem LocalVersionedEngine::append_internal(
     }
 }
 
+VersionedItem LocalVersionedEngine::refresh_symbol_cache() {
+    struct SymbolData {
+        StreamId stream_id_;
+        timestamp first_index_;
+        timestamp last_index_;
+        size_t num_rows_;
+    };
+
+    auto stream_ids = symbol_list().get_symbols(store());
+    std::vector<VersionQuery> version_queries;
+    version_queries.resize(stream_ids.size());
+    auto versions = batch_get_versions(store(), version_map(), stream_ids, version_queries);
+    std::vector<folly::Future<SymbolData>> output;
+    for (auto &&version : folly::enumerate(versions)) {
+        output.emplace_back(std::move(*version).thenValue([store = store()](auto maybe_index_key) -> ReadKeyOutput {
+            util::check(static_cast<bool>(maybe_index_key), "Version not found for stream");
+            return store->read_sync(maybe_index_key.value());
+        }).thenValue([store = store()](auto &&key_segment_pair) {
+            auto [index_key, index_segment] = std::move(key_segment_pair);
+            using FieldType = pipelines::index::Fields;
+            const auto first_index = pipelines::index::index_start_from_segment<SegmentInMemory, FieldType>(index_segment, 0);
```

Comment:
This is intentionally not reading the descriptor right? So won't return timezone information?

Reply:
We could add a timezone column to the output dataframe if required.

Reply:
But then we'll need to read the stream descriptor?

Reply:
The encoded field changes will make that all faster, we can choose to decode just the protobuf part which is small and fast, or the whole descriptor including columns etc. The problem at the moment is it's all or nothing

Reply:
So I don't think we can do it with this PR until the other changes are merged, or it will be unacceptably slow